addons/payment/models/payment_acquirer.py

```diff
@@ -304,155 +279,121 @@ class PaymentTransaction(osv.Model):
-            values.update(self.on_change_partner_id(cr, uid, None, values.get('partner_id'), context=context)['value'])
+    @api.model
+    def create(self, values):
+        PaymentAcquirer = self.env['payment.acquirer']
 
         # call custom create method if defined (i.e. ogone_create for ogone)
         if values.get('acquirer_id'):
-            acquirer = self.pool['payment.acquirer'].browse(cr, uid, values.get('acquirer_id'), context=context)
+            acquirer = PaymentAcquirer.browse(values.get('acquirer_id'))
 
             # compute fees
             custom_method_name = '%s_compute_fees' % acquirer.provider
-            if hasattr(Acquirer, custom_method_name):
-                fees = getattr(Acquirer, custom_method_name)(
-                    cr, uid, acquirer.id, values.get('amount', 0.0), values.get('currency_id'), values.get('partner_country_id'), context=None)
-                values['fees'] = float_round(fees, 2)
+            if hasattr(acquirer, custom_method_name):
+                fees = getattr(acquirer, custom_method_name)(values.get('amount', 0.0), values.get('currency_id'), values.get('partner_country_id'))
+                values['fees'] = tools.float_round(fees[0], 2)
 
             # custom create
             custom_method_name = '%s_create' % acquirer.provider
             if hasattr(self, custom_method_name):
-                values.update(getattr(self, custom_method_name)(cr, uid, values, context=context))
+                values.update(getattr(self, custom_method_name)(values))
 
         # Default value of reference is
-        tx_id = super(PaymentTransaction, self).create(cr, uid, values, context=context)
+        transaction = super(PaymentTransaction, self).create(values)
         if not values.get('reference'):
-            self.write(cr, uid, [tx_id], {'reference': str(tx_id)}, context=context)
-        return tx_id
+            transaction.write({'reference': str(transaction.id)})
+        return transaction
 
-    def write(self, cr, uid, ids, values, context=None):
-        Acquirer = self.pool['payment.acquirer']
+    @api.multi
+    def write(self, values):
+        Acquirer = self.env['payment.acquirer']
         if ('acquirer_id' in values or 'amount' in values) and 'fees' not in values:
             # The acquirer or the amount has changed, and the fees are not explicitely forced. Fees must be recomputed.
-            if isinstance(ids, (int, long)):
-                ids = [ids]
-            for txn_id in ids:
+            for transaction in self:
                 vals = dict(values)
                 vals['fees'] = 0.0
-                transaction = self.browse(cr, uid, txn_id, context=context)
                 if 'acquirer_id' in values:
-                    acquirer = Acquirer.browse(cr, uid, values['acquirer_id'], context=context) if values['acquirer_id'] else None
+                    acquirer = Acquirer.browse(values['acquirer_id']) if values['acquirer_id'] else None
```

Comment:
Please @gpa-odoo check the return value of ***browse*** I find comment meaningless : 
```
browse return False not None
```